packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2006,13 +2043,22 @@ class _SubmenuButtonState extends State<SubmenuButton> {
           ),
         );
 
-        if (_enabled && _platformSupportsAccelerators) {
-          return MenuAcceleratorCallbackBinding(
-            onInvoke: () => toggleShowMenu(context),
-            hasSubmenu: true,
+        if (_enabled) {
```

Comment:
Maybe invert this and just return child here if !_enabled. Simplifies the later logic.